rustls/src/server/server_conn.rs

```diff
@@ -1051,6 +1055,13 @@ impl EarlyDataState {
         matches!(self, Self::Rejected)
     }
 
+    fn peek(&self) -> Option<&[u8]> {
+        match self {
```

Comment:
Nit: I think it would be more idiomatic to use match ergonomics (`match &self { .. }` and drop the `ref` keyword).